eventlet/green/nanomsg.py

```diff
@@ -0,0 +1,56 @@
+# coding: utf-8
+"""The :mod:`nanomsg` module wraps the :class:`Socket`
+found in :mod:`nanomsg-python <nanomsg>` to be non blocking.
+"""
+
+__nanomsg__ = __import__('nanomsg')
+from eventlet.patcher import slurp_properties
+from eventlet.hubs import trampoline
+from nanomsg import wrapper
+
+__nanomsg__.__all__ = ['wrapper', 'NanoMsgError', 'NanoMsgAPIError', 'Device', 'Socket']
+
+for name, value in wrapper.nn_symbols():
+    if name.startswith('NN_'):
+        name = name[3:]
+    __nanomsg__.__all__.append(name)
+
+__patched__ = ['Socket']
+slurp_properties(__nanomsg__, globals(), ignore=__patched__)
```

Comment:
Do those two blocks above mutate global state on import?

Reply:
They do. Unfortunately nanomsg doesn't provide an ``__all__``, which is required by ``slurp_properties()``, so the monkeypatcher builds one for it.

Reply:
I've submitted a PR to tonysimpson/nanomsg-python#71 to add an ``__all__`` variable to the nanomsg module.

Reply:
It would be even better if your code defers to their `__all__` once it exists upstream.